### ISSUE TITLE
Cut 0.4.0 Release

### ISSUE THREAD

**Issue by jlewi** [kubeflow/kubeflow#2098]
Tracking bug for the 0.4 release.

**Comment by jlewi** [comment]
Here are some outstanding PRs that should be merged before cutting the release

#2060 - Custom image support in Jupyter UI
#2080 - Upgrade cloud endpoints controller

**Comment by IronPan** [comment]
Pipeline need a new release to be included with sample fixes. Create an issue for tracking
https://github.com/kubeflow/pipelines/issues/549

**Comment by Jeffwan** [comment]
- [ ] #2067 Enforce use string type param for S3_USE_HTTPS and S3_VERIFY_SSL
- [ ] #2097 Update aws parameters in serving and dashboard
- [ ] #2073 Use upper method from std library on v0.3 branch .(I need a another patch for master)

**Comment by Jeffwan** [comment]
@jlewi @r2d4 Thanks a lot on the help of above PRs. All merged into master branch and the left one failed tests due to same argo workflow test failures on 0.3-branch. Since it's not related to 0.4-release,  I will try to figure it out later. Appreciate that!

**Comment by johnugeorge** [comment]
- [ ] #2126 Update ksonnet package for Katib

**Comment by xyhuang** [comment]
- [ ] #2110 Ks package for kubebench operator

**Comment by jlewi** [comment]
Have we enabled periodic tests on the release branch for 0.4 yet?

**Comment by jlewi** [comment]
I did a quick sanity check of 0.4 based on
https://github.com/kubeflow/kubeflow/commit/8799d29cd172752138ad502ae0c59afcff7c8a87

* Checked in the spawner UI I can specify a custom image
* All the central dashboard links are working.

One issue I saw was it looks like the Jupyter images for TF 1.11 & 1.12 aren't included in the default config.

**Comment by jlewi** [comment]
according to @IronPan 0.4 release branch is using the latest version of pipelines.

**Comment by r2d4** [comment]
/close
https://github.com/kubeflow/kubeflow/tree/v0.4.0
